Lib/_weakrefset.py

```diff
@@ -90,7 +90,13 @@ def clear(self):
         self.data.clear()
 
     def copy(self):
-        return self.__class__(self)
+        new = WeakSet()
```

Comment:
I'm not sure that it works as expected for subclasses.

Reply:
I could change this to be:
```suggestion
        new = self.__class__()
```

but then it would be different to the other weak collections

Reply:
would it be better to use `return self.__class__(self)` in all the weak collections and then in `.update()` special case instances of `__class__`?

Reply:
Other WeakRef collections predate WeakSet. The latter's been added in 3b8cb176952 and used `self.__class__(self)` for copying ever since. It's invalid now to change this to `WeakSet()` in the interest of "consistency".

Changing other weak collections to use `self.__class__(self)` is also technically a backwards incompatible change but with less opportunity for breakage. That we could definitely consider.

Reply:
Ok I'll switch them over and special case update